src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeLoader.cs

```diff
@@ -34,5 +43,48 @@ public ResourceTypeComponents LoadType(ResourceTypeReference reference)
 
             return resourceTypeFactory.GetResourceType(serializedResourceType);
         }
+
+        public ResourceTypeComponents? LoadFallbackResourceType()
+        {
+            if (fallbackResourceType != null)
+            {
+                var serializedResourceType = typeLoader.LoadResourceType(fallbackResourceType);
+
+                return resourceTypeFactory.GetResourceType(serializedResourceType);
+            }
+
+            // No fallback type provided in JSON
+            return null;
+        }
+
+        public NamespaceConfiguration? LoadNamespaceConfiguration()
+        {
+            if (typeSettings != null)
```

Comment:
I think it's fine to throw here if `typeSettings` is null - we don't want providers to publish without it.

It might be worth adding a follow up task to ensure it's set in `bicep publish`.

Reply:
I added the settings object to the index files that are referenced for `bicep publish provider`